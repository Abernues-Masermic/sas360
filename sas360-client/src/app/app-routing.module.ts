import { CheckLoginGuard } from './shared/guards/check-login.guard';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
  {
    path: '',
    canActivate: [CheckLoginGuard],
    loadChildren: () =>
      import('./pages/home/home.module').then(m => m.HomeModule),
  },
  {
    path: 'notFound',
    loadChildren: () =>
      import('./pages/not-found/not-found.module').then(m => m.NotFoundModule),
  },
  {
    path: 'admin',
    canActivate: [CheckLoginGuard],
    loadChildren: () =>
      import('./pages/admin/admin.module').then(m => m.AdminModule),
  },
  {
    path: 'login',
    loadChildren: () =>
      import('./pages/auth/login/login.module').then(m => m.LoginModule),
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
