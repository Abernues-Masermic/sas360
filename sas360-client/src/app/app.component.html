<app-spinner></app-spinner>
<mat-sidenav-container>
  <mat-sidenav #sidenav mode="side" [(opened)]="opened">
    <mat-toolbar color="primary">ADMIN</mat-toolbar>
    <app-sidebar></app-sidebar>
  </mat-sidenav>

  <mat-sidenav-content>
    <app-header (toogleSidenav)="sidenav.toggle()"></app-header>
    <router-outlet></router-outlet>
    <app-footer></app-footer>
  </mat-sidenav-content>
</mat-sidenav-container>
<hr />
