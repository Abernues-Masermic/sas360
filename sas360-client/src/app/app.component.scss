
@import 'variables';

mat-toolbar.mat-primary {
  background: $nav-color;
  color: $white;
}

mat-sidenav{
  width:200px;
}

mat-sidenav-content{
  background-color: $main-color;
}

mat-sidenav-container{
  height: 100%;
}

