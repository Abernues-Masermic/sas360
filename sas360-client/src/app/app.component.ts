import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
import { ChildrenOutletContexts } from '@angular/router';
import { Subject, takeUntil } from 'rxjs';

import { AuthService } from '@pagesauth/auth.service';
import { UtilsService } from '@shared/services/utils.service';
import { UserResponse } from '@shared/models/user.interface';
import { PAGE_SLIDE_ANIMATION } from '@shared/animations/pagesAnimation';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  animations: [PAGE_SLIDE_ANIMATION],
})
export class AppComponent implements OnInit, OnDestroy {
  title = 'sas360-client';
  openned = false;
  private destroy$ = new Subject<any>();

  constructor(
    private contexts: ChildrenOutletContexts,
    private readonly utilsSvc: UtilsService,
    private readonly authSvc: AuthService
  ) {}

  ngOnInit(): void {
    this.utilsSvc.sidebarOpenned$
      .pipe(takeUntil(this.destroy$))
      .subscribe((res: boolean) => (this.openned = res));

    this.authSvc.user$
      .pipe(takeUntil(this.destroy$))
      .subscribe((userresponse: UserResponse | null) => {
        if (userresponse == null) {
          this.openned = false;
        }
      });
  }

  ngOnDestroy(): void {
    this.destroy$.next({});
    this.destroy$.complete();
  }

  getRouteAnimationData() {
    return this.contexts.getContext('primary')?.route?.snapshot?.data?.[
      'animation'
    ];
  }
}
