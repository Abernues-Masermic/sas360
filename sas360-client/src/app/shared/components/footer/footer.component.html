<footer class="footer">
  <div class="content">
    <p>Copyright &copy; 2021 Masermic</p>
    <a routerLink="/about">About</a>
  </div>
</footer>
