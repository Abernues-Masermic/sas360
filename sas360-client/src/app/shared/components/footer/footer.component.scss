@import 'variables';

.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 60px;
  background: linear-gradient(90deg, rgb(0, 0, 0) 15%, rgba(47, 47, 48, 0.782) 52%, rgb(68, 68, 68) 100%);
  color: $white;
  text-align: center;

  .content{
    font-family: $app-font-family;
    font-size: $app-font-size;
    margin-top: $app-margin-top;
    margin-bottom: $app-margin-bottom;

    a{
      color: $white;
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }
}
