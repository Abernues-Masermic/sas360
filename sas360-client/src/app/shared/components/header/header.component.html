<mat-toolbar color="primary">
  <ng-container *ngIf="userRole === 'ADMIN'">
    <button mat-icon-button (click)="onToogleSideNav()">
      <mat-icon>menu</mat-icon>
    </button>
  </ng-container>

  <span>SAS360</span>
  <span class="horizontal-spacer"></span>

  <h3>{{ userName }}</h3>
  <a *ngIf="!isLogged; else viewLogout">
    <a class="custom-button-class" routerLink="/login">Login</a>
  </a>
</mat-toolbar>

<ng-template #viewLogout>
  <a class="custom-button-class" (click)="onLogout()">Logout</a>
</ng-template>
