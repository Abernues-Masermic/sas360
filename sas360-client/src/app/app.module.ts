import { NgModule } from '@angular/core';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { ToastrModule } from 'ngx-toastr';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { MaterialModule } from '@shared/material.module';
import { SpinnerInterceptor } from '@shared/interceptors/spinner.interceptor';
import { HeaderComponent } from '@shared/components/header/header.component';
import { FooterComponent } from '@shared/components/footer/footer.component';
import { PopUpComponent } from '@shared/components/pop-up/pop-up.component';
import { SidebarModule } from '@shared/components/sidebar/sidebar.module';
import { SpinnerModule } from '@shared/components/spinner/spinner.module';

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
    PopUpComponent,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    BrowserAnimationsModule,
    SidebarModule,
    MaterialModule,
    HttpClientModule,
    ReactiveFormsModule,
    CommonModule,
    FormsModule,
    SpinnerModule,
    ToastrModule.forRoot(),
  ],

  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: SpinnerInterceptor, multi: true },
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
