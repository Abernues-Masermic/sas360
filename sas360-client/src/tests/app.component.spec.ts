import { SidebarModule } from './../app/shared/components/sidebar/sidebar.module';
import { FooterComponent } from '@shared/components/footer/footer.component';
import { HeaderComponent } from '@shared/components/header/header.component';
import { MaterialModule } from '@shared/material.module';
import {
  BrowserAnimationsModule,
  NoopAnimationsModule,
} from '@angular/platform-browser/animations';
import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';
import { HttpClient } from '@angular/common/http';
import { UtilsService } from '@shared/services/utils.service';
import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '@app/app.component';

describe('AppComponent', () => {
  let httpClient: HttpClient;
  let httpTestingController: HttpTestingController;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppComponent, HeaderComponent, FooterComponent],
      imports: [
        MaterialModule,
        HttpClientTestingModule,
        RouterTestingModule,
        BrowserAnimationsModule,
        NoopAnimationsModule,
        SidebarModule,
      ],
      providers: [UtilsService],
    }).compileComponents();
    httpClient = TestBed.inject(HttpClient);
    httpTestingController = TestBed.inject(HttpTestingController);
  });

  afterEach(() => {
    httpTestingController.verify();
  });

  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  test(`should have as title 'sas360-client'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component.title).toEqual('sas360-client');
  });

  test('debe hacer match con el snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled).toMatchSnapshot();
  });
});
