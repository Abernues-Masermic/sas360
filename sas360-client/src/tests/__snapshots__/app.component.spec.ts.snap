// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent debe hacer match con el snapshot 1`] = `
<div
  id="root2"
>
  <mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container"
  >
    <div
      class="mat-drawer-backdrop"
    />
    
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav ng-trigger ng-trigger-transform mat-drawer-side"
      mode="side"
      style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
      tabindex="-1"
    >
      <div
        cdkscrollable=""
        class="mat-drawer-inner-container"
      >
        <mat-toolbar
          class="mat-toolbar mat-primary mat-toolbar-single-row"
          color="primary"
        >
          ADMIN
        </mat-toolbar>
        <app-sidebar>
          <div
            class="vertical-spacer"
          />
          <mat-nav-list
            class="mat-nav-list mat-list-base"
            role="navigation"
          >
            <a
              class="mat-list-item mat-focus-indicator"
              href="/admin"
              mat-list-item=""
              routerlink="/admin"
            >
              <span
                class="mat-list-item-content"
              >
                <span
                  class="mat-ripple mat-list-item-ripple"
                  mat-ripple=""
                />
                <span
                  class="mat-list-text"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate white-icon material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  insert_chart_outlined
                </mat-icon>
                <span
                  class="nav-link"
                >
                  Dashboard
                </span>
              </span>
            </a>
            <a
              class="mat-list-item mat-focus-indicator"
              href="/admin/users"
              mat-list-item=""
              routerlink="/admin/users"
            >
              <span
                class="mat-list-item-content"
              >
                <span
                  class="mat-ripple mat-list-item-ripple"
                  mat-ripple=""
                />
                <span
                  class="mat-list-text"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate white-icon material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  people
                </mat-icon>
                <span
                  class="nav-link"
                >
                  Users
                </span>
              </span>
            </a>
            <a
              class="mat-list-item mat-focus-indicator"
              href="/admin/devices"
              mat-list-item=""
              routerlink="/admin/devices"
            >
              <span
                class="mat-list-item-content"
              >
                <span
                  class="mat-ripple mat-list-item-ripple"
                  mat-ripple=""
                />
                <span
                  class="mat-list-text"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate white-icon material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  settings_overscan
                </mat-icon>
                <span
                  class="nav-link"
                >
                  Devices
                </span>
              </span>
            </a>
            <a
              class="mat-list-item mat-focus-indicator"
              href="/admin/events"
              mat-list-item=""
              routerlink="/admin/events"
            >
              <span
                class="mat-list-item-content"
              >
                <span
                  class="mat-ripple mat-list-item-ripple"
                  mat-ripple=""
                />
                <span
                  class="mat-list-text"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate white-icon material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  settings_overscan
                </mat-icon>
                <span
                  class="nav-link"
                >
                  Events
                </span>
              </span>
            </a>
            <a
              class="mat-list-item mat-focus-indicator"
              mat-list-item=""
            >
              <span
                class="mat-list-item-content"
              >
                <span
                  class="mat-ripple mat-list-item-ripple"
                  mat-ripple=""
                />
                <span
                  class="mat-list-text"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate white-icon material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  exit_to_app
                </mat-icon>
                <span
                  class="nav-link"
                >
                  Exit
                </span>
              </span>
            </a>
          </mat-nav-list>
        </app-sidebar>
      </div>
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    >
      <app-header>
        <mat-toolbar
          class="mat-toolbar mat-primary mat-toolbar-single-row"
          color="primary"
        >
          
          <span>
            SAS360
          </span>
          <span
            class="horizontal-spacer"
          />
          <h3>
            
          </h3>
          <a>
            <a
              class="customButton"
              href="/login"
              routerlink="/login"
            >
              Login
            </a>
          </a>
          
        </mat-toolbar>
        
      </app-header>
      <router-outlet />
      
      <app-footer>
        <footer
          class="footer"
        >
          <div
            class="content"
          >
            <p>
              Copyright Â© 2021 Masermic
            </p>
            <a
              href="/about"
              routerlink="/about"
            >
              About
            </a>
          </div>
        </footer>
      </app-footer>
    </mat-sidenav-content>
    
  </mat-sidenav-container>
  <hr />
</div>
`;
